### PR TITLE
implement first iteration of json reports

### PR DESCRIPTION
**Pull request by iduuck**

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const logError = require('./utils/logError')
 program
   .version(pkg.version)
   .option('-t, --threshold <number>', 'Specify a custom threshold')
+  .option('-P, --pretty', 'Prettify the output (only with `json` reporter)')
```
Comment (visualcookie) [inline bin/index.js:10]: Option to prettify is not working. Returning:
```
$ flow-cov --reporters json --pretty
error: unknown option `--pretty'
error Command failed with exit code 1.
```

**Review by visualcookie — CHANGES_REQUESTED** [review]
While having reporters defined, it returns an unhandled promise rejection warning.
```
$ flow-cov --reporters json
An error happened, while executing `flow-cov`:
Error: Reporter found, that is not supported.

(node:7861) UnhandledPromiseRejectionWarning: Error: Error: Reporter found, that is not supported.
    at logError (/Users/xxx/Fintory/xxx/node_modules/flow-cov/bin/utils/logError.js:31:9)
    at run (/Users/xxx/Fintory/xxx/node_modules/flow-cov/bin/index.js:41:12)
    at <anonymous>
(node:7861) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:7861) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by iduuck** [comment]
Can you please give me the content of your `lib/reporters/index.js`. Also could you verify the existance of `lib/reporters/json.js`?

**Comment by visualcookie** [comment]
> Can you please give me the content of your `lib/reporters/index.js`. Also could you verify the existance of `lib/reporters/json.js`?

Content of `lib/reporters/index.js`:
``` javascript
const textReporter = require('./text')
const jsonReporter = require('./json')

module.exports = function launchReporter(type, report) {
  switch (type) {
    case 'json':
      return jsonReporter(report)
    case 'text':
      return textReporter(report)
    default:
      throw new Error('Reporter found, that is not supported.')
  }
}
```

Can also confirm the existence of `lib/reporters/json.js` with the following content:
``` javascript
module.exports = function jsonReporter(report) {
  const contents = JSON.stringify(report, null, report.config.pretty ? 2 : 0)

  console.log(contents)

  return {
    reporterName: 'json',
    contents,
  }
}
```

**Comment by iduuck** [comment]
OK! Strange one.

Can you set the DEBUG flag? Like `DEBUG=true flow-cov`? And post the log?

**Comment by visualcookie** [comment]
> OK! Strange one.
> 
> Can you set the DEBUG flag? Like `DEBUG=true flow-cov`?

That's strange. With that flag it worked now. Command run: `flow-cov --reporters json --pretty`.

Even after removing the DEBUG again.

**Review by visualcookie — APPROVED** [review]

**Pull request merged**